### PR TITLE
Fix missing mpm_prefork on FreeBSD when using itk.(httpd: Configurati…

### PR DESCRIPTION
**Pull request by jsm222**
…on error: No MPM loaded)

### CODE REVIEW

File: manifests/mod/itk.pp [hunk 1]
```diff
@@ -19,6 +19,7 @@
     }
   } else {
     # prefork is a requirement for itk in 2.4; except on FreeBSD and Gentoo, which are special
```
Comment (smortex) [inline manifests/mod/itk.pp]: This line could be fixed to illustrate what you are saying (« …except on Gentoo, which is… »)

File: manifests/mod/itk.pp [hunk 2]
```diff
@@ -19,6 +19,7 @@
...
+    # Note FreeBSD still requires prefork module see below.
```
Comment (smortex) [inline manifests/mod/itk.pp]: … then this comment is not needed anymore …

File: manifests/mod/itk.pp [hunk 3]
```diff
@@ -19,6 +19,7 @@
...
     if $::osfamily =~ /^(FreeBSD|Gentoo)/ {
```
Comment (smortex) [inline manifests/mod/itk.pp]: … then this could be fixed by removing *FreeBSD* …

File: manifests/mod/itk.pp [hunk 4]
```diff
@@ -79,6 +80,13 @@
       apache::mpm{ 'itk':
         apache_version => $apache_version,
       }
+		# On FreeBSD mpm_prefork_module is required together with itk or you will get httpd: "Configuration error: No MPM loaded".
+		if ($::osfamily =~ /FreeBSD/) {
+		File { "$::apache::mod_dir/mpm_prefork.load":
+				content=>"LoadModule mpm_prefork_module $::apache::lib_path/mod_mpm_prefork.so",
+				ensure => 'present'
+	        }
+      	}
```
Comment (smortex) [inline manifests/mod/itk.pp]: …and then this whole addition should not be necessary :-)  (beware of the indentation: you used tabs instead of spaces!)

**Comment by smortex** [comment]
Hey @jsm222, I added a few inline comments, can you please update your pull request to take this into account?

Thanks!

File: manifests/mod/itk.pp (hunk 3 above)
Comment (smortex) [inline manifests/mod/itk.pp]: A trailing `$` would also be appropriate IMHO : `/^(Gentoo)$/`

**Comment by jsm222** [comment]
I used the addition otherwise I got a double declaration of apache::package. 
I actually started out like your suggestion by simply removing FreeBSD from  [the regex](https://github.com/smortex/puppetlabs-apache/pull/1#discussion-diff-73965583R23)
That gives `Evaluation Error: Error while evaluating a Resource Statement, Duplicate declaration: Class[Apache::Package] is already declared in file /usr/local/etc/puppet/environments/production/modules/apache/manifests/mpm.pp:87; cannot redeclare at /usr/local/etc/puppet/environments/production/modules/apache/manifests/mpm.pp:87 at /usr/local/etc/puppet/environments/production/modules/apache/manifests/mpm.pp:87:7 at /usr/local/etc/puppet/environments/production/modules/apache/manifests/mod/itk.pp:79 on node` I have tried to fix it by simplifying package.pp even more, and added some of the FreeBSD package logic in mpm.pp. Not tested on BSD with apache ver < 2.4

**Pull request closed without merging**